src/main/java/com/wepay/zktools/zookeeper/internal/ZooKeeperClientImpl.java

```diff
@@ -54,7 +57,25 @@
 
     private volatile boolean running = true;
 
+    /**
+     * Constructor to create a ZooKeeperClientImpl instance.
+     * @param connectString comma separated host:port pairs, each corresponding to a zk server.
+     * @param sessionTimeout ZooKeeper session timeout in milliseconds.
+     * @throws ZooKeeperClientException exception to be thrown if connection failure.
+     * */
     public ZooKeeperClientImpl(String connectString, int sessionTimeout) throws ZooKeeperClientException {
+        this(connectString, sessionTimeout, null);
+    }
+
+    /**
+     * Avoid directly use this constructor with a not-null connectTimeout parameter, unless you purposefully want to
+     * timeout the connection to ZooKeeper servers and exit retrying when there's no ZooKeeper running.
```

Comment:
this statement can be avoided I feel.

Reply:
So our concern here is that zktools is supposed to always manage the zookeeper session lifecycle for applications, as advised in zktools readme. So it's probably good to suggest users that such usage would potentially break that. But maybe it's not necessary, what do you think?

Reply:
@smithakoduri updated with more general javadoc, PTAL